index/test/rtree/rtree_with_strategies.cpp

```diff
@@ -0,0 +1,156 @@
+// Boost.Geometry Index
+// Unit Test
+
+// Copyright (c) 2025 Adam Wulkiewicz, Lodz, Poland.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+
+#define BOOST_GEOMETRY_INDEX_DETAIL_EXPERIMENTAL_PREDICATES
+
+#include <geometry_index_test_common.hpp>
+
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/geometries/register/box.hpp>
+#include <boost/geometry/geometries/register/segment.hpp>
+
+struct point { double x, y; };
+struct box { point mi, ma; };
+struct segment { point first, second; };
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(point, double, cs::undefined, x, y)
+BOOST_GEOMETRY_REGISTER_BOX(box, point, mi, ma)
+BOOST_GEOMETRY_REGISTER_SEGMENT(segment, point, first, second)
+
+template <typename QueryValue, typename Rtree>
+constexpr bool is_bp_or_bb_v = bg::util::is_box<QueryValue>::value
+                               && bg::util::is_box<typename Rtree::value_type>::value
+                            || bg::util::is_box<QueryValue>::value
+                               && bg::util::is_point<typename Rtree::value_type>::value;
```

Comment:
Everywhere I work it is required to put brackets, there were `&&` and `||` are combined. Also compilers normally warn for it. Can we do it?
Though the layout is beautiful here, so it's not unclear to me.
And it's test code.

Nit, bp or bb (but the order is: bb or bp)

Reply:
Sure, done.